### PR TITLE
Fix full outer join output row count

### PR DESCRIPTION
**Pull request by dirtysalt**
ref: #2124 

Considering it's very hard to make up this case in UT, and fix is quite straightforwad, is it okay not to add UT?

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java [hunk 1]
```diff
@@ -742,7 +742,7 @@ private Void computeJoinNode(ExpressionContext context, ScalarOperator joinOnPre
                 builder.setOutputRowCount(max(0, rightRowCount - innerRowCount));
                 break;
             case FULL_OUTER_JOIN:
-                builder.setOutputRowCount(leftRowCount + rightRowCount - innerRowCount);
+                builder.setOutputRowCount(max(0, leftRowCount + rightRowCount - innerRowCount));
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java:745]: leftRowCount + rightRowCount should always large than `innerRowCount`, The real error seems happen before

Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java:745]: I think max(0, leftRowCount + rightRowCount - innerRowCount) is better

**Review by Youngwb — DISMISSED** [review]

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Comment by dirtysalt** [comment]
run starrocks_clang-format

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**